[ci] run stylelint as part of CI tests

christian-byrne opened the pull request:
## Summary

Final PR continuing from

- https://github.com/Comfy-Org/ComfyUI_frontend/pull/5926
- https://github.com/Comfy-Org/ComfyUI_frontend/pull/5940

actually run stylelint in lint workflows.

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-6143-ci-run-stylelint-as-part-of-CI-and-pre-commit-hook-2916d73d3650811891bcc9f94379cc5f) by [Unito](https://www.unito.io)

christian-byrne committed:
invoke stylelint in ci

DrJKL reviewed (changes requested):
Yes please, but only for the CI part.

christian-byrne committed:
[refactor] remove stylelint from pre-commit hook to improve performance - addresses review feedback

Stylelint remains in CI workflow for quality checks while reducing
pre-commit hook overhead that was causing developers to bypass hooks.

Co-authored-by: DrJKL <DrJKL@users.noreply.github.com>

christian-byrne commented on .github/workflows/ci-lint-format.yaml:
```diff
@@ -54,6 +54,9 @@ jobs:
       - name: Run ESLint with auto-fix
         run: pnpm lint:fix
 
+      - name: Run Stylelint with auto-fix
```
Thanks for the suggestion. Keeping the workflow without additional comments for now to maintain consistency with other workflow steps.

christian-byrne commented on lint-staged.config.js:
```diff
@@ -1,9 +1,20 @@
     'pnpm typecheck'
+  ],
+
+  './**/*.vue': (stagedFiles) => [
```
Vue file separation was removed in 2885e8c to address pre-commit performance concerns. Vue files now run with the standard TypeScript pipeline without stylelint in pre-commit (stylelint remains in CI only).

christian-byrne commented on packages/design-system/src/css/style.css:
```diff
@@ -258,9 +258,15 @@
-  --color-interface-menu-component-surface-hovered: var(--interface-menu-component-surface-hovered);
-  --color-interface-menu-component-surface-selected: var(--interface-menu-component-surface-selected);
-  --color-interface-menu-keybind-surface-default: var(--interface-menu-keybind-surface-default);
+  --color-interface-menu-component-surface-hovered: var(
```
Agreed\! Stylelint's value-keyword-case and indentation rules handle these formatting improvements automatically for better readability.

christian-byrne commented on packages/design-system/src/css/style.css:
```diff
@@ -1139,7 +1145,7 @@ audio.comfy-audio.empty-audio-widget {
 
 .isLOD .lg-node-header {
-  border-radius: 0px;
+  border-radius: 0;
```
Thanks\! Stylelint's length-zero-no-unit rule removes unnecessary units from zero values automatically.
